zerver/lib/actions.py

```diff
@@ -1298,6 +1298,9 @@ def do_send_messages(messages_maybe_none: Sequence[Optional[MutableMapping[str,
         message['message'].rendered_content_version = bugdown_version
         links_for_embed |= message['message'].links_for_preview
 
+        if message['message'].is_stream_message():
+            message['message'].rendering_realm = message['realm']
```

Comment:
We should probably set this unconditionally, as it's useful elsewhere.  

I'm on board with doing the work to make this a globally available concept.  I think we should call it `message.realm`, though, and write a migration to add it to all historical messages.

Reply:
What should we be using as the default value? For historical messages, does it make sense to stick to `message.sender.realm` or should we write logic to figure out the stream/PMed users from the Recipient data and get the stream's realm?

Reply:
We need to write that sort of migration logic to look it up, whihc will be kinda annoying.